pkg/controller/master/mcmsettings/fleetcontroller.go

```diff
@@ -0,0 +1,134 @@
+package mcmsettings
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	mgmtv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/wrangler/pkg/relatedresource"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+
+	"github.com/harvester/harvester/pkg/util"
+)
+
+// watchFleetControllerConfigMap will watch the configmap for fleet-controller setting and requeue internal-server-url and internal-cacerts to trigger reconcile of associated settings to ensure correct values are applied
+func (h *mcmSettingsHandler) watchFleetControllerConfigMap(_, _ string, obj runtime.Object) ([]relatedresource.Key, error) {
+	if cm, ok := obj.(*corev1.ConfigMap); ok {
+		if cm.Name == util.DefaultFleetControllerConfigMapName && cm.Namespace == util.DefaultFleetControllerConfigMapNamespace {
+			return []relatedresource.Key{
+				{
+					Name: util.RancherInternalCASetting,
+				},
+				{
+					Name: util.RancherInternalServerURLSetting,
+				},
+			}, nil
+		}
+	}
+	return nil, nil
+}
+
+func (h *mcmSettingsHandler) watchInternalEndpointSettings(_ string, setting *mgmtv3.Setting) (*mgmtv3.Setting, error) {
+	if setting == nil || setting.DeletionTimestamp != nil || (setting.Name != util.RancherInternalCASetting && setting.Name != util.RancherInternalServerURLSetting) {
+		return setting, nil
+	}
+
+	logrus.Debugf("reconcilling fleet-controller config for setting %s", setting.Name)
+
+	fleetControllerConfig, err := h.configMapCache.Get(util.DefaultFleetControllerConfigMapNamespace, util.DefaultFleetControllerConfigMapName)
+	if err != nil {
```

Comment:
Please consider the posssible future case:

When the upstream Rancher does not use the default namespace/name. Here we may use h.configMap.Get() and check if the error is IsNotExisting, then log a warning and return nil.

Reply:
the current fleet-controller design needs the config map to exist. If this changes we can accordingly update or remove this handler if this is not needed.